components/script/dom/htmlelement.rs

```diff
@@ -839,10 +904,97 @@ impl VirtualMethods for HTMLElement {
                     DOMString::from(&**attr.value()),
                 );
             },
+            (&local_name!("form"), mutation) => {
+                if self.is_form_associated_custom_element() {
```

Comment:
These conditions can be merged:

```suggestion
            (&local_name!("form"), mutation) if self.is_form_associated_custom_element() {
```